From c1435a1ffc41b58db24c2855288967bc01b2b546 Mon Sep 17 00:00:00 2001
From: Artem Gindinson <artem.gindinson@intel.com>
Date: Tue, 13 Dec 2022 14:10:57 +0000
Subject: [PATCH]  Allow LLVM 12+ min/max intrinsics in IGC IR

As a follow-up to intel/intel-graphics-compiler@2882458, let integer min/max
intrinsics through IGC IR Verifier to handle some cases with LLVM 14/15 (in
particular, those coming from compute runtime tests).
---
 IGC/Compiler/IGC_IR_spec.hpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/IGC/Compiler/IGC_IR_spec.hpp b/IGC/Compiler/IGC_IR_spec.hpp
index c8d25e9c8a7..7c9f7730379 100644
--- a/IGC/Compiler/IGC_IR_spec.hpp
+++ b/IGC/Compiler/IGC_IR_spec.hpp
@@ -102,6 +102,10 @@ IGC_IR_LLVM_INTRINSIC(sadd_sat)
 #endif
 #if LLVM_VERSION_MAJOR >= 12
 IGC_IR_LLVM_INTRINSIC(abs)
+IGC_IR_LLVM_INTRINSIC(smax)
+IGC_IR_LLVM_INTRINSIC(smin)
+IGC_IR_LLVM_INTRINSIC(umax)
+IGC_IR_LLVM_INTRINSIC(umin)
 IGC_IR_LLVM_INTRINSIC(experimental_noalias_scope_decl)
 #endif
 
